tests/parsing/GlobbingTest.py

```diff
@@ -313,9 +313,11 @@ def test_collect_recursive(self):
         self._test_glob(pattern, file_list)
 
     def test_collect_recursive_part_of_basename(self):
-        pattern = os.path.join(TestFiles.glob_test_dir, "**.py")
+        pattern = os.path.join(TestFiles.glob_test_dir, "**.(py|[xy])")
```

Comment:
While something like `*.([xy])` (which should match `file.x` and `file.y` I think) is going to be rare, we should add a test just in case IMO.